### DISCUSSION TITLE
How to Automatically Update Schema When Database Has No Relations but Schema Has Manual Relations

### DISCUSSION THREAD

**Discussion by Lindsay0214** [prisma/prisma#25179] [Q&A]
### Question

Our NestJS project using Prisma with MariaDB, the database was set up using an old Java project, and it lacks primary and foreign key relationships. However, our Prisma schema has defined these relationships, so we cannot synchronize the database and generate the new schema through migrations, we would have to handle this manually. I'm concern about potential errors with manual handling and we preference for automation (e.g., through migrations). What would be the best approach in this situation?

We only thought of generating the SQL scripts for what needs to be done today and then running them as scripts. We would repeat this process each time we need to make new updates.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @Lindsay0214 👋 

These are the steps you can use to handle this scenario:

1. Introspect the Existing Database: Use [introspection](https://www.prisma.io/docs/orm/prisma-schema/introspection) to pull the current state of your database schema into your Prisma schema file. This will help you get a baseline of your existing database structure.
```
npx prisma db pull
```
2. Modify the Prisma Schema: Update your Prisma schema file to include the necessary primary and foreign key relationships that are currently missing in your database.

3. Generate the Migration: Create a new migration based on the updated Prisma schema.
```
npx prisma migrate dev --create-only
```
4. Review the Migration Script: Review the generated SQL migration script to ensure it correctly adds the primary and foreign keys. Make any necessary adjustments to the script.

5. Apply the Migration: Apply the migration to your database. This will update your database schema to match the Prisma schema.
```
npx prisma migrate dev
```
Do let us know if you run into any issues while doing this.

**Reply by Lindsay0214** [reply]
Yes, but because we cannot update the existing database, if our database does not have primary and foreign key relationships while our Prisma schema does, this could cause issues.

So now I've thought of a method: perhaps we can add columns by creating a view table? This way, it won't affect the existing table.

Otherwise, we would ultimately have to create a new database, but since we are assisting with the frontend refactoring, this is scheduled for the next phase...
